### ISSUE TITLE
Feature: dark mode

### ISSUE THREAD

**Issue by legallyfree** [ipfs/ipfs-webui#1702]
**Is your feature request related to a problem? Please describe.**
My eyes, I go blind at night when I see the screen, this window is very bright.

**Describe the solution you'd like**
They could place a selection of themes to change the colors to suit one.

**Describe alternatives you've considered**
use the colors used by gtk / kde apps and use those colors / themes

**Additional context**
try to change the dependency of the use of npm for yarn, I use yarn instead of npm but well it is not so important but I must mention it since npm is not the only one.

**Comment by jessicaschilling** [comment]
Additional context for anyone who comes across this issue:
- We'd very much like to include a dark-mode option in Web UI (and, by extension, Desktop)
- Current timeline would involve including this work in a larger-scale revisiting of Web UI/Desktop, likely in Q2/3 2021
- If anyone is interested in PRing something earlier, contributions are most welcome!

**Comment by lidel** [comment]
If someone wants to work on this:
- speed up your work by copying colors generated  by https://darkreader.org for now
  - we can refine and bubble up them to https://github.com/ipfs-shipyard/ipfs-css later

**Comment by KaKi87** [comment]
Hello,
Any news on this ?
Thanks

**Comment by voidvoxel** [comment]
Photosensitive user here; I sadly just had to uninstall the desktop app due to the blindingly-bright screen with no dark mode option. Support for dark mode would be greatly appreciated! I'll be sticking to the CLI for now :blush:

**Comment by Artoria2e5** [comment]
The colors are pretty spread out in the js files. Guess a theme-switch button will really be needed -- no easy path with only changing a CSS file with `@media (prefers-color-scheme)`. Well, I mean, you can do the same query with JS, but if you're changing the JS you might as well add a button.

**Comment by kaf-lamed-beyt** [comment]
> If someone wants to work on this:
> 
> * speed up your work by copying colors generated  by https://darkreader.org for now
>   
>   * we can refine and bubble up them to https://github.com/ipfs-shipyard/ipfs-css later

hi @lidel, can you assign this to me so I can attempt it

**Comment by lidel** [comment]
Sure, feel free to work on this and ping when PR is  ready for review.

**Comment by lidel** [comment]
In case its useful for testing, you can force brave and google chrome to run in dark mode via `--force-dark-mode` i think

**Comment by kaf-lamed-beyt** [comment]
Alright, I'll do this too

**Comment by lidel** [comment]
#2322  is a good start but missed some key parts of UI that makes it unusable on systems with Dark Mode preference.

@SgtPooki @kaf-lamed-beyt  Should revert until below are addressed, or is the plan to address below before next release?

1. status page → graph contrast is too high
   > ![Image](https://github.com/user-attachments/assets/a907e6db-ffb8-4395-b8f0-d3712a418175)
1. status page → Tutorial:
   > ![Image](https://github.com/user-attachments/assets/479c6fe1-4a38-4a2a-b25f-c57220f92816)
1. Status page → Addresses 
   > ![Image](https://github.com/user-attachments/assets/e33be86a-4d3b-4570-b4e9-1fa74661e387)
1. Context menus and all modals on Files screen:
   > ![Image](https://github.com/user-attachments/assets/5f3dc794-bd8d-413f-bfde-6b9fbd0ef0d0)
   > ![Image](https://github.com/user-attachments/assets/f4036ce7-6c50-4751-967f-b018e9bd9702)
   > ![Image](https://github.com/user-attachments/assets/5df5713c-2634-47d2-9b8e-a4d9ca26578b)
   > etc.
1. Explore landing screen has white box:
   > ![Image](https://github.com/user-attachments/assets/7ef7afd7-9a6b-49b6-92e6-9141c74f95d2)
1. Exploring DAG is unreadable:
   > ![Image](https://github.com/user-attachments/assets/6e1f7e0b-3bbd-4b5a-94d8-8e7d436cbaa5)
1. Peers screen → Add connection modal is unreadable 
   > ![Image](https://github.com/user-attachments/assets/360811b2-71c3-4be3-ba6e-ee2e89bd1895)
1. Settings screen → Add pinning service and IPNS modals are unreadable:
   > ![Image](https://github.com/user-attachments/assets/184607b3-6726-47be-a8d8-47dcc004e745)
   > ![Image](https://github.com/user-attachments/assets/2b627ad9-c137-4fd2-9f37-170f7e738500)

**Comment by kaf-lamed-beyt** [comment]
Ah! Crap! I missed all those modals and context menus. 

I'll try opening a new PR for these, @lide, if that's okay. It mostly just updates to the respective styles if I'm not mistaken. 

For the addresses UI, I couldn't reproduce that. Is there a way you got the addresses to show? Do I have to include some type of input or somn'? 

Kindly walk me through, thanks.

**Comment by SgtPooki** [comment]
FYI I reverted the dark mode changes so if you could throw that back together with the additional fixes that would be great.

> For the addresses UI, I couldn't reproduce that. Is there a way you got the addresses to show? Do I have to include some type of input or somn'?

The addresses should show after your node has been running for a little bit and then you just click the little `^ Advanced` toggle to show the other windows

**Comment by kaf-lamed-beyt** [comment]
Definitely. I'll get to it, @SgtPooki

**Comment by kaf-lamed-beyt** [comment]
@SgtPooki, I'm done with the updates... I'm guessing I'd need to create a new PR, yeah?

@lidel, looking at the explore page... I think it depends on this package `ipld-explorer-components/pages`. I couldn't find the exact components rendering the UI on that route and on `/explore/[cid]`

**Comment by lidel** [comment]
@kaf-lamed-beyt ipld-explorer-components live in https://github.com/ipfs/ipld-explorer-components/ – open a companion PR in that repo.

**Comment by kaf-lamed-beyt** [comment]
alright, thanks! I'll set that up.

**Comment by kaf-lamed-beyt** [comment]
but, wait... @lidel, how would you suggest I go about this? Will I include the theme context I already have here in that repo? how does it work? can you walk me through?

**Comment by SgtPooki** [comment]
> but, wait... [@lidel](https://github.com/lidel), how would you suggest I go about this? Will I include the theme context I already have here in that repo? how does it work? can you walk me through?

@kaf-lamed-beyt I think what we would need to do is add some CSS to ipld-explorer-components with a parent selector of `[data-theme='dark']` to handle the display.. and we can leave it at that.

**Comment by kaf-lamed-beyt** [comment]
alright, that sounds good @SgtPooki. I'll get to it and let you know if I run into any blocker(s).

**Comment by kaf-lamed-beyt** [comment]
@SgtPooki, I've opened a [PR](https://github.com/ipfs/ipld-explorer-components/pull/471) in the explorer-components repo.

**Comment by lidel** [comment]
Triage notes:
- Initial implementation from  https://github.com/ipfs/ipfs-webui/pull/2322  was reverted in https://github.com/ipfs/ipfs-webui/pull/2339 due to bugs described in https://github.com/ipfs/ipfs-webui/issues/1702#issuecomment-2686363521
- If anyone is interested in working on this, the above may be useful starting point

**Comment by ChayanDass** [comment]
Hi @lidel   , i want to take it up !

**Comment by kushmunjal** [comment]
Hi @lidel , I have been searching about this problem for quiet some time because this issue would also fix another issue with ipfs-desktop repo and started the work for this can I take this issue up , can I be assigned to it
 
thanks

**Comment by volkkov** [comment]
Good catch. This is related to the RFC change in how trailing slashes are normalized. PR incoming.

**Comment by kushmunjal** [comment]
<img width="2499" height="1279" alt="Image" src="https://github.com/user-attachments/assets/fd575133-fbdf-4869-a814-8a54966312dc" />

<img width="2504" height="1297" alt="Image" src="https://github.com/user-attachments/assets/230e9b37-4308-4e2a-b938-1f55094b615d" />

I've opened a PR for this issue: #2496 .
check it out !
Summary:

- Added a three-way theme system (Auto, Light, and Dark).
- Auto follows the OS prefers-color-scheme and updates live when the system theme changes.
- Added theme switching from both the top bar and the Settings page.
- Replaced hardcoded colors with centralized theme-aware CSS variables across the UI to improve consistency and maintainability.
- Fixed the Settings page themeDescription translation issue.
- Manually verified the application across pages, modals, dropdowns, and selection states in both light and dark themes.

Any feedback, suggestions, or review comments are more than welcome—I’d be happy to iterate and improve the implementation.

Happy coding!

**Comment by kushmunjal** [comment]
**@lidel** please review my pr and can you give me feedback for this issue it would help me in my open source contribution journey

Thanks
